RocketChat/Rocket.Chat
# feat: apply group filter to LDAP sync process

## Proposed changes (including videos or screenshots)
Use the group filter to filter the LDAP integration and sync only the users that match the criteria.

Today we sync all the users from the Base DN and use the group filter only to restrict the ones that can access the RC. The goal is to only sync the users in the group filter setting to avoid importing unwanted users and consuming unnecessary seats.





## Issue(s)


## Steps to test or reproduce
1. Create a `groupOfNames` in the LDAP;
2. Use the `cn` attribute to define the group name;
3. Set in the `member` attribute the user that you want to have access to the RC (create multiple `member` attributes if you need multiple users);
4. In Rocket.chat, navigate to workspace > settings > LDAP > User Search;
5. Enable the Group Filter and fill the fields;
6. Sync the data and check the users page;
7. Try to login with a user that is in the RC database, but not in the filtered group;


## Further comments

[SUP-305](https://rocketchat.atlassian.net/browse/SUP-305)


[SUP-305]: https://rocketchat.atlassian.net/browse/SUP-305?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
